test/control-plane/agent.go

```diff
@@ -0,0 +1,115 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package controlplane
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+
+	fakeApiExt "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/fake"
+	"k8s.io/client-go/kubernetes/fake"
+
+	"github.com/cilium/cilium/daemon/cmd"
+	fakeDatapath "github.com/cilium/cilium/pkg/datapath/fake"
+	"github.com/cilium/cilium/pkg/endpoint"
+	fqdnproxy "github.com/cilium/cilium/pkg/fqdn/proxy"
+	"github.com/cilium/cilium/pkg/k8s"
+	fakeCilium "github.com/cilium/cilium/pkg/k8s/client/clientset/versioned/fake"
+	fakeSlim "github.com/cilium/cilium/pkg/k8s/slim/k8s/client/clientset/versioned/fake"
+	"github.com/cilium/cilium/pkg/k8s/version"
+	"github.com/cilium/cilium/pkg/node/types"
+	"github.com/cilium/cilium/pkg/option"
+	"github.com/cilium/cilium/pkg/proxy"
+)
+
+type agentHandle struct {
+	d       *cmd.Daemon
+	tempDir string
+}
+
+func (h *agentHandle) tearDown() {
+	h.d.Close()
+	os.RemoveAll(h.tempDir)
+}
+
+func startCiliumAgent(nodeName string, clients fakeClients, modConfig func(*option.DaemonConfig)) (*fakeDatapath.FakeDatapath, agentHandle, error) {
+	types.SetName(nodeName)
+	k8s.Configure("dummy", "dummy", 10.0, 10)
+	version.Force("1.23")
+
+	k8s.SetClients(clients.core, clients.slim, clients.cilium, clients.apiext)
+
+	proxy.DefaultDNSProxy = fqdnproxy.MockFQDNProxy{}
+	option.Config.Populate()
+	option.Config.IdentityAllocationMode = option.IdentityAllocationModeCRD
+	option.Config.DryMode = true
+	option.Config.Opts = option.NewIntOptions(&option.DaemonMutableOptionLibrary)
+	option.Config.Opts.SetBool(option.DropNotify, true)
+	option.Config.Opts.SetBool(option.TraceNotify, true)
+	option.Config.Opts.SetBool(option.PolicyVerdictNotify, true)
+	option.Config.Opts.SetBool(option.Debug, true)
+	option.Config.EnableIPSec = false
+	option.Config.EnableIPv6 = false
+	option.Config.KubeProxyReplacement = option.KubeProxyReplacementStrict
+	option.Config.EnableHostIPRestore = false
+	option.Config.K8sRequireIPv6PodCIDR = false
+	option.Config.K8sEnableK8sEndpointSlice = true
+	option.Config.EnableL7Proxy = false
+	option.Config.EnableHealthCheckNodePort = false
+	option.Config.Debug = true
```

Comment:
We have all of these options, that's fine. Now, how can I run my tests if they depend on `EnableIPv6` to be `true` for example?